File: yandex_new_year_adventures_2025/3. Мандарины и апельсины/solution.cs
//Wrong tests. Look at the python solution and compare to this solution

using System;
using System.Collections.Generic;
using System.Linq;


class Program 
{
    static void Main() 
    {
        int n = int.Parse(Console.ReadLine());
        List<(long Mandarins, long Oranges)> boxes = new List<(long, long)>();
        long totalMandarins = 0;
        long totalOranges = 0;

        for (int i = 0; i < 2 * n - 1; i++)
        {
            string[] line = Console.ReadLine().Split();
            long mandarins = long.Parse(line[0]);
            long oranges = long.Parse(line[1]);
            boxes.Add((mandarins, oranges));
            totalMandarins += mandarins;
            totalOranges += oranges;
        }

        boxes.Sort((a, b) => (b.Mandarins + b.Oranges).CompareTo(a.Mandarins + a.Oranges));
        

        long currentMandarins = 0;
        long currentOranges = 0;

        for (int i = 0; i < boxes.Count; i++) // we need n boxes? but tests need all boxes
        {
            currentMandarins += boxes[i].Mandarins;
            currentOranges += boxes[i].Oranges;

            if (currentMandarins >= totalMandarins / 2 && currentOranges >= totalOranges / 2)
            {
                Console.WriteLine("Yes");
                return;
            }
        }

        Console.WriteLine("No");
    }
}

File: yandex_new_year_adventures_2025/3. Мандарины и апельсины/solution_.py
import sys
from itertools import combinations

def can_choose_boxes(n, boxes):
    total_mandarins = sum(box[0] for box in boxes)
    total_oranges = sum(box[1] for box in boxes)
    
    boxes.sort(key=lambda x: (-x[0], -x[1]))
    
    current_mandarins = 0
    current_oranges = 0
    
    for box in boxes:
        current_mandarins += box[0]
        current_oranges += box[1]
        
        if current_mandarins >= total_mandarins // 2 and current_oranges >= total_oranges // 2:
            return "Yes"
    
    return "No"

def main():
    n = int(input())
    fruits = []
    for _ in range(2*n-1):
        mandarins, oranges = map(int, input().split())
        fruits.append((mandarins, oranges))

    print(can_choose_boxes(n, fruits))

if __name__ == '__main__':
    main()